Add rw612 apsta support

Liam-qk opened the pull request:
drivers: wifi: nxp: Set default bandwidth for softap
    
    When params->bandwidth is 0, set it as a default value(WIFI_FREQ_BANDWIDTH_20MHZ).

samples: net: wifi: Add apsta_mode support for frdm_rw612 and rd_rw612_bga.
    
    Add conf files for frm_rw612 and rd_rw612_bga to support apsta_mode example.

samples: net: wifi: Add new Kconfig for apsta_mode

     Introduce a dedicated Kconfig option for apsta_mode example to allow more flexible configuration in the application layer.
    This change helps make apsta_mode more generic and easier to integrate across different vendor platforms.
     
     Add compile-time assertions for the following macros:
    CONFIG_WIFI_SAMPLE_AP_SSID
    CONFIG_WIFI_SAMPLE_SSID
    CONFIG_WIFI_SAMPLE_AP_IP_ADDRESS
    CONFIG_WIFI_SAMPLE_AP_NETMASK
    
    Default to empty string ("") if the following
    two macros are not defined:
    CONFIG_WIFI_SAMPLE_AP_PSK
    CONFIG_WIFI_SAMPLE_PSK

samples: net: wifi: Enable apsta_mode macros in *.conf file.

    Update *.conf to enable apsta_mode macros without impacting origin boards support.

samples: net: wifi: Add apsta_mode configurations for rw612.

    Add more apsta_mode configurations for frdm_rw612 and rd_rw612_bga.

Liam-qk committed:
samples: net: wifi: Add apsta_mode support for rw612 boards.

Add conf files for frm_rw612 and rd_rw612_bga
to support apsta_mode example.

Signed-off-by: Qiankun Li <qiankun.li@nxp.com>

Liam-qk committed:
drivers: wifi: nxp: Set default bandwidth for softap

When params->bandwidth is 0, set it
as a default value(WIFI_FREQ_BANDWIDTH_20MHZ).

Signed-off-by: Qiankun Li <qiankun.li@nxp.com>

Liam-qk committed:
samples: net: wifi: Add new Kconfig for apsta_mode

Introduce a dedicated Kconfig option for apsta_mode example to
allow more flexible configuration in the application layer.
This change helps make apsta_mode more generic and easier to
integrate across different vendor platforms.

Add compile-time assertions for the following macros:
CONFIG_WIFI_SAMPLE_AP_SSID
CONFIG_WIFI_SAMPLE_SSID
CONFIG_WIFI_SAMPLE_AP_IP_ADDRESS
CONFIG_WIFI_SAMPLE_AP_NETMASK

Default to empty string ("") if the following
two macros are not defined:
CONFIG_WIFI_SAMPLE_AP_PSK
CONFIG_WIFI_SAMPLE_PSK

Signed-off-by: Qiankun Li <qiankun.li@nxp.com>

Liam-qk committed:
samples: net: wifi: Add apsta_mode configurations for rw612.

Add more apsta_mode configurations for frdm_rw612
and rd_rw612_bga.

Add overlay file to set apsta_mode configurations.

Signed-off-by: Qiankun Li <qiankun.li@nxp.com>

jukkar commented on samples/net/wifi/apsta_mode/Kconfig:
```diff
@@ -0,0 +1,49 @@
+
+config WIFI_AP_SSID
+	string "SSID of soft AP"
+	default "SOFT-AP"
```
Please remove all the defaults here and below as these are related to the user configuration and are not generic settings that all users would use.

jukkar commented on samples/net/wifi/apsta_mode/src/main.c:
```diff
@@ -41,11 +31,11 @@ static void wifi_event_handler(struct net_mgmt_event_callback *cb, uint64_t mgmt
 	switch (mgmt_event) {
 	case NET_EVENT_WIFI_CONNECT_RESULT: {
-		LOG_INF("Connected to %s", WIFI_SSID);
+		LOG_INF("Connected to %s", CONFIG_WIFI_SSID);
```
Add build asserts so that the compilation fails if user has not set the config option values that need to be set.

jukkar commented on samples/net/wifi/apsta_mode/src/main.c:
```diff
@@ -118,21 +110,23 @@ static int enable_ap_mode(void)
-	ap_config.psk = (const uint8_t *)WIFI_AP_PSK;
-	ap_config.psk_length = strlen(WIFI_AP_PSK);
+	ap_config.ssid = (const uint8_t *)CONFIG_WIFI_AP_SSID;
+	ap_config.ssid_length = strlen(CONFIG_WIFI_AP_SSID);
```
Instead of `strlen` do `sizeof(CONFIG_....) - 1` so that the value is created at built time.

jukkar commented on samples/net/wifi/apsta_mode/boards/frdm_rw612.conf:
```diff
@@ -4,6 +4,9 @@ CONFIG_ETH_DRIVER=n
 CONFIG_WIFI_NM=y
 CONFIG_NXP_WIFI_SOFTAP_SUPPORT=y
 
+CONFIG_WIFI_AP_IP_ADDRESS="192.168.10.1"
```
This cannot be here as now someone having this board will get this IP address by default which is probably not something user wants.
Also the DHCPv4 option should be set somewhere else that needs to disable the setting.

jukkar commented on samples/net/wifi/apsta_mode/boards/rd_rw612_bga.conf:
```diff
@@ -3,6 +3,9 @@ CONFIG_NXP_RW610=y
 CONFIG_WIFI_NM=y
 CONFIG_NXP_WIFI_SOFTAP_SUPPORT=y
 
+CONFIG_WIFI_AP_IP_ADDRESS="192.168.10.1"
```
same here

Liam-qk committed:
samples: net: wifi: Enable apsta_mode macros in *.conf file.

Update *.conf to enable apsta_mode macros
without impacting origin boards support.

Signed-off-by: Qiankun Li <qiankun.li@nxp.com>

jukkar commented on samples/net/wifi/apsta_mode/boards/frdm_rw612.conf:
```diff
@@ -4,6 +4,13 @@ CONFIG_ETH_DRIVER=n
 CONFIG_WIFI_NM=y
 CONFIG_NXP_WIFI_SOFTAP_SUPPORT=y
 
+# apsta_mode configuration
+CONFIG_WIFI_SAMPLE_AP_SSID="FRDM_RW612_AP"
+CONFIG_WIFI_SAMPLE_AP_PSK=""
+CONFIG_WIFI_SAMPLE_SSID="TEST_AP"
+CONFIG_WIFI_SAMPLE_PSK="PASSWORD"
+CONFIG_WIFI_SAMPLE_DHCPV4_START=n
```
Do you need these values for NXP specific tests that are run by your CI? I have this board and if I compile this sample for it, then I will the default values that do not really reflect my setup. Best would be that these configurations are specified in NXP downstream repo, but if you really want to have these in upstream, we could do this in similar way as in wifi shell where NXP specific config files are placed to nxp sub-directory (see `samples/net/wifi/shell/nxp` for details)

Liam-qk replied:
Okay, will add an overlay file similar to samples/net/wifi/shell/nxp/*.conf.

jukkar commented on samples/net/wifi/apsta_mode/boards/rd_rw612_bga.conf:
```diff
@@ -3,6 +3,13 @@ CONFIG_NXP_RW610=y
 CONFIG_WIFI_NM=y
 CONFIG_NXP_WIFI_SOFTAP_SUPPORT=y
 
+# apsta_mode configuration
+CONFIG_WIFI_SAMPLE_AP_SSID="RD_RW612_BGA_AP"
+CONFIG_WIFI_SAMPLE_AP_PSK=""
+CONFIG_WIFI_SAMPLE_SSID="TEST_AP"
+CONFIG_WIFI_SAMPLE_PSK="PASSWORD"
+CONFIG_WIFI_SAMPLE_DHCPV4_START=n
```
same comment for these

jukkar commented on samples/net/wifi/apsta_mode/socs/esp32_procpu.conf:
```diff
@@ -3,3 +3,12 @@ CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4=y
 CONFIG_WIFI_NM=y
 CONFIG_WIFI_NM_MAX_MANAGED_INTERFACES=2
+
+# apsta_mode configuration
```
This makes the sample application work the same way as before which I suppose is okeish, but the configuration should not be hardcoded in the application that affects all users of this soc. At some point these settings should be removed from this global file and be placed to an overlay file instead so that users wanting these values can use them. FYI @sylvioalves

MaochenWang1 commented:
Hi @jukkar @sylvioalves please review
